include/SDL3/SDL_time.h

```diff
@@ -0,0 +1,244 @@
+extern DECLSPEC int SDLCALL SDL_DateTimeToTime(const SDL_DateTime *dt, SDL_Time *ticks);
+
+/**
+ *
+ * Given an SDL_DateTime struct, check if the members are within valid ranges to represent a real date
+ * and time.
+ *
+ * \param dt the SDL_DateTime
+ * \returns SDL_TRUE if all struct members are within valid ranges, otherwise SDL_FALSE; call SDL_GetError()
+ *          for more information.
+ *
+ * \since This function is available since SDL 3.0.0
+ */
+extern DECLSPEC SDL_bool SDLCALL SDL_DateTimeIsValid(const SDL_DateTime *dt);
+
+/* Converts an SDL time into a Windows FILETIME (100-nanosecond intervals since January 1, 1601).
+ *
+ * This function fills in the two 32-bit values of the FILETIME structure.
+ *
+ * \param ticks the time to convert
+ * \param dwLowDateTime a pointer filled in with the low portion of the Windows FILETIME value
+ * \param dwHighDateTime a pointer filled in with the high portion of the Windows FILETIME value
+ *
+ * \since This function is available since SDL 3.0.0.
+ */
+extern DECLSPEC void SDLCALL SDL_TimeToWindows(SDL_Time ticks, Uint32 *dwLowDateTime, Uint32 *dwHighDateTime);
+
+/* Converts a Windows FILETIME (100-nanosecond intervals since January 1, 1601) to an SDL time
+ *
+ * This function takes the two 32-bit values of the FILETIME structure as parameters.
+ *
+ * \param dwLowDateTime the low portion of the Windows FILETIME value
+ * \param dwHighDateTime the high portion of the Windows FILETIME value
+ * \returns the converted SDL time
+ *
+ * \since This function is available since SDL 3.0.0.
+ */
+extern DECLSPEC SDL_Time SDLCALL SDL_TimeFromWindows(Uint32 dwLowDateTime, Uint32 dwHighDateTime);
+
+/**
+ *
+ * Get the number of days in a month for a given year.
+ *
+ * \param year the year
+ * \param month the month [1-12]
+ * \returns the number of days in the requested month, otherwise -1; call SDL_GetError() for more information.
+ *
+ * \since This function is available since SDL 3.0.0
+ */
+extern DECLSPEC int SDLCALL SDL_GetDaysInMonth(int year, int month);
+
+/**
+ *
+ * Get the day of year for a calendar date.
+ *
+ * \param year the year component of the date
+ * \param month the month component of the date
+ * \param day the day component of the date
+ * \returns the day of year [0-365] if the date is valid, otherwise -1; call SDL_GetError()
+ *          for more information.
+ *
+ * \since This function is available since SDL 3.0.0
+ */
+extern DECLSPEC int SDLCALL SDL_GetDayOfYear(int year, int month, int day);
+
+/**
+ *
+ * Get the day of week for a calendar date.
+ *
+ * \param year the year component of the date
+ * \param month the month component of the date
+ * \param day the day component of the date
+ * \returns a value between 0 and 6 (0 being Sunday) if the date is valid, otherwise -1; call SDL_GetError()
+ *          for more information.
+ *
+ * \since This function is available since SDL 3.0.0
+ */
+extern DECLSPEC int SDLCALL SDL_GetDayOfWeek(int year, int month, int day);
```

Comment:
I'm not going to question this one, I've wished I had it before. :)

Reply:
We have the enumeration SDL_DAY_OF_WEEK, should we return it here, and maybe add an error value to the enumeration?

Reply:
That's a good idea.

Reply:
Or maybe not, if we want to interoperate with the C standard library, which takes an int, I presume?

Reply:
The enum values match the C standard week day numbers, so at worst, it would require a cast.

Reply:
Yeah, it would require a cast. One of my goals for SDL is to minimize the casting people have to do when using the API. Let's leave it as int for now, and document that 0 starts the week at Sunday.